collector/lib/CollectorConfigInspector.cpp

```diff
@@ -0,0 +1,61 @@
+#include "CollectorConfigInspector.h"
+
+#include <forward_list>
+#include <string>
+#include <unordered_map>
+
+#include <arpa/inet.h>
+
+#include <google/protobuf/util/json_util.h>
+
+#include "ConnTracker.h"
+
+namespace collector {
+
+const std::string CollectorConfigInspector::kBaseRoute = "/state/config";
+
+CollectorConfigInspector::CollectorConfigInspector(const std::shared_ptr<CollectorConfig> config) : config_(config) {
+}
+
+Json::Value CollectorConfigInspector::configToJson() {
+  Json::Value root;
+  std::optional<sensor::CollectorConfig> runtime_config = config_->GetRuntimeConfig();
+
+  if (runtime_config.has_value()) {
+    std::string jsonString;
+    const auto& config = runtime_config.value();
+    google::protobuf::util::Status status = google::protobuf::util::MessageToJsonString(config, &jsonString);
+
+    if (!status.ok()) {
+      CLOG(WARNING) << "Failed to convert protobuf object to JSON: " << status.ToString();
+      return Json::Value();
+    }
+
+    Json::CharReaderBuilder readerBuilder;
+    std::string errs;
+    std::istringstream iss(jsonString);
+    if (!Json::parseFromStream(readerBuilder, iss, &root, &errs)) {
+      CLOG(ERROR) << "Failed to parse JSON string: " << errs;
+      return Json::Value();
+    }
```

Comment:
If the caller to this method is just going to grab the `Json::Value` object and turn it into a string, why are we grabbing a string and turning it into a `Json::Value` here? Why can't we just return the string?

Reply:
Done